src/provider/provider_tracking.h

```diff
@@ -10,18 +10,61 @@
 #ifndef UMF_MEMORY_TRACKER_INTERNAL_H
 #define UMF_MEMORY_TRACKER_INTERNAL_H 1
 
+#include <assert.h>
+
 #include <umf/base.h>
 #include <umf/memory_pool.h>
 #include <umf/memory_provider.h>
 
+#include "critnib.h"
+#include "utils_concurrency.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 extern umf_memory_provider_ops_t UMF_TRACKING_MEMORY_PROVIDER_OPS;
 
+struct umf_memory_tracker_t {
+    critnib *map;
+    os_mutex_t *splitMergeMutex;
```

Comment:
Global split/merge lock might kill scalability. As I understand the lock is needed only when concurrent split/merge occurs for the same memory buffer.

Reply:
well, mmap is also fully synchronized, so I wouldn't worry that much. This shouldn't be a super common operation. I'd start with a simple mutex and then we can think about locking out ranges of memory if there's a perf problem.